src/main/kotlin/uk/gov/dluhc/printapi/service/CertificateBatchingService.kt

```diff
@@ -29,12 +35,40 @@ class CertificateBatchingService(
 
     fun batchCertificates(batchSize: Int): Map<String, List<Certificate>> {
         val certificatesPendingAssignment = certificateRepository.findByStatus(Status.PENDING_ASSIGNMENT_TO_BATCH)
-        return certificatesPendingAssignment.chunked(batchSize).associate { batchOfCertificates ->
+
+        return limitCertificates(certificatesPendingAssignment).chunked(batchSize).associate { batchOfCertificates ->
             val batchId = idFactory.batchId()
             batchId to batchOfCertificates.onEach {
                 it.getCurrentPrintRequest().batchId = batchId
                 it.addStatus(Status.ASSIGNED_TO_BATCH)
             }
         }
     }
+
+    private fun limitCertificates(certificatesPendingAssignment: List<Certificate>): List<Certificate> {
+        val startOfDay = Instant.now(clock).truncatedTo(DAYS)
+        val endOfDay = startOfDay.plus(1, DAYS).minusSeconds(1)
+        val countOfRequestsSentToPrintProvider =
+            certificateRepository.getPrintRequestStatusCount(startOfDay, endOfDay, Status.ASSIGNED_TO_BATCH)
+
+        if (certificatesPendingAssignment.size + countOfRequestsSentToPrintProvider > dailyLimit) {
+            logDailyLimit(certificatesPendingAssignment, countOfRequestsSentToPrintProvider)
+            return certificatesPendingAssignment.subList(0, dailyLimit - countOfRequestsSentToPrintProvider)
+        }
+
+        return certificatesPendingAssignment
+    }
+
+    private fun logDailyLimit(
+        certificatesPendingAssignment: List<Certificate>,
+        countOfRequestsSentToPrintProvider: Int
+    ) {
+        logger.warn {
+            """Identified ${certificatesPendingAssignment.size} certificates to assign to a batch. 
+            Daily print limit is $dailyLimit. 
+            $countOfRequestsSentToPrintProvider certificates already sent to print provider today.
+            Remaining capacity is ${dailyLimit - countOfRequestsSentToPrintProvider}.
+            ${certificatesPendingAssignment.size + countOfRequestsSentToPrintProvider - dailyLimit} certificates won't be assigned to a batch."""
```

Comment:
Minor, might be worth changing the final sentence very slightly to make it clear the certs that wont be batched will be batched tomorrow - its not that they wont be batched ever.
Something like:
```
... won't be assigned to a batch today.
```
or
```
... won't be assigned to a batch until tomorrow.
```
or
```
... won't be assigned to a batch until <tomorrow date> at the earliest.
```
(where <tomorrow date> is an ISO8601 of tomorrow 00:0:00 - eg: `2022-11-25T00:00:00.000Z` - might be more complex than its worth though, so happy if you decide not to do this)

Reply:
I guess in theory we will write this message every time the batching task runs after the daily limit is reached.
Technically we are counting certificates that have been added to a batch - if there has been problems writing to SFTP or more generally the zipping task fails... we won't have "sent to print provider today".

Remaining capacity seems a bit misleading as after the current batch is processed there is no remaining capacity.
Maybe could simplify to something like:
```
  Daily print limit of X reached.  
  X certificates pending assignment will be processed tomorrow.
````

Reply:
Thanks, @NathanRussellValtech ,
I agree, the current log message might be misleading. I will implement the second option you proposed.

Thanks, @Neil-Massey , for your suggestion as well. 
I might have missed your point, but the remaining capacity in the message refers to the number of requests that can be added to a batch as part of the current batching task. When it runs the next time, and the daily limit is reached, the remaining capacity will be presented as 0.

Reply:
When I read the message the remaining capacity is always zero - so didn't think including it was useful.